src/main/java/seedu/address/ui/entitylistpanel/EntityListPanel.java

```diff
@@ -1,14 +1,110 @@
 package seedu.address.ui.entitylistpanel;
 
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+
 import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import seedu.address.model.entity.Entity;
+import seedu.address.model.entity.Mentor;
+import seedu.address.model.entity.Participant;
+import seedu.address.model.entity.PrefixType;
+import seedu.address.model.entity.Team;
+import seedu.address.ui.EntityCard;
 import seedu.address.ui.UiPart;
 
 /**
  * Panel containing the list of persons.
  */
-public abstract class EntityListPanel extends UiPart<Region> {
+public class EntityListPanel extends UiPart<Region> {
+    private static final String FXML = "EntityListPanel.fxml";
+    private PrefixType prefix;
+
+    @FXML
+    private ListView<Entity> entityListView;
+    @FXML
+    private VBox panelContainer;
+
+    //TODO: clarify
+    //Not sure whether these generics is advisable, please advise
+    public EntityListPanel(ObservableList<? extends Entity> entityList) {
```

Comment:
Hey I'm not too great with generics, but maybe this is the sourceof the problem u mentioned in the telegram chat? Would it be possible to just put ObservableList<Entity> instead?

Reply:
Hey Brian, I solved the issue, thank you for the help :)